packages/dappkit/src/index.ts

```diff
@@ -72,41 +125,170 @@ async function getGasCurrencyContract(
 }
 
 export interface TxParams<T> {
-  txId: string
   tx: TransactionObject<T>
   from: string
   to: string
   gasCurrency: GasCurrency
+  estimatedGas?: number
+  value?: string
 }
 
 export async function requestTxSig<T>(
   web3: Web3,
   txParams: TxParams<T>[],
   meta: DappKitRequestMeta
 ) {
+  // TODO: For multi-tx payloads, we for now just assume the same from address for all txs. We should apply a better heuristic
+  const baseNonce = await web3.eth.getTransactionCount(txParams[0].from)
   const txs: TxToSignParam[] = await Promise.all(
-    txParams.map(async (txParam) => {
+    txParams.map(async (txParam, index) => {
       const gasCurrencyContract = await getGasCurrencyContract(web3, txParam.gasCurrency)
+      const value = txParam.value === undefined ? '0' : txParam.value
+
       const estimatedTxParams = {
         gasCurrency: gasCurrencyContract.options.address,
+        from: txParam.from,
+        value,
       }
-      // @ts-ignore
-      const estimatedGas = await txParam.tx.estimateGas(estimatedTxParams)
+      const estimatedGas =
+        txParam.estimatedGas === undefined
+          ? //
+            // @ts-ignore
```

Comment:
do `estimatedTxParams as any` or something else, instead of `ts-ignore`
This potentially hides problems when we factor things.
Remember, every time we write a `ts-ignore` a puppy dies :(
We need to erradicate them